### ISSUE TITLE
Transaction subscription does not seem to notify on outgoing txs for p2sh addresses

### ISSUE THREAD

**Issue by mr-zwets** [gcash/bchd#495]
The transaction subscription of the grpc does not seem to notify the client on outgoing transactions for a p2sh address set in the filter.
Incoming transactions to the p2sh address do work.

**Comment by zquestz** [comment]
Good catch, and thanks for the report. The behaviour is real, though it turned out not to be P2SH specific.

Subscriptions matched inputs only against outpoints the filter had already seen arrive as outputs. Nothing seeded that map when you subscribed, so spending anything the address held before the subscription started went unnoticed, for any address type.

P2PKH looked fine because wallets spending from an address usually send change back to it. The notification was firing on the change output rather than on the spend. A contract spend from a P2SH address often has no output back to the address, so nothing matched, which is why it showed up there first.

Fixed: subscriptions now resolve the output an input is spending and match on the address that funded it, so outgoing transactions notify regardless of whether the filter saw the funding output. One caveat, resolving a confirmed parent needs the transaction index, so nodes running without `--txindex` will still only match spends of outputs observed during the subscription.
